### PR TITLE
[Designate] Add Mdns replica count

### PR DESCRIPTION
**Pull request by omersch381**
DesignateMdns deployment type is being changed to stateful set.

This commit adds replica count to Mdns in all the Designate CRs.

### CODE REVIEW

**Comment by omersch381** [comment]
https://github.com/openstack-k8s-operators/designate-operator/pull/255 will have to get merged first.

**Review by jpodivin — CHANGES_REQUESTED** [review]
Hi. Could you add an explicit depends on to this PR? Just so it's obvious that it's missing a dependency. 

Also, it would be great to have a link to JIRA item, or some other description of what are you implementing and why. Thanks a lot.

**Comment by omersch381** [comment]
> me other description of what are you implementing and why. Thanks a lot.

Sure, thanks for raising it.

**Comment by omersch381** [comment]
The Designate-operator patch has been merged yesterday, and it added the Replicas field to the API directory files, I wonder when CI will stop failing because of lacking it.

**Review by bshephar — APPROVED** [review]

File: config/samples/base/openstackcontrolplane/core_v1beta1_openstackcontrolplane.yaml [hunk 1]
```diff
@@ -205,7 +205,7 @@ spec:
       designateProducer:
         replicas: 1 # backend needs to be configured
       designateMdns:
-        replicas: 0 # backend needs to be configured
+        replicas: 1 # backend needs to be configured
```
Comment (dprince) [inline config/samples/base/openstackcontrolplane/core_v1beta1_openstackcontrolplane.yaml:208]: comment here says backend needs to be configured still. If you are setting replicas to 1 does this (and if functionally works) does this comment still make sense?

Comment (omersch381) [reply config/samples/base/openstackcontrolplane/core_v1beta1_openstackcontrolplane.yaml:208]: It looks like it is a generic comment that appear on many files. We are still configuring MiniDNS backend on the designate-operator repository, so I think it still applies.
This openstack-operator patch mainly saves us time, so we won't have to modify the csv manually with the same values. We modify its value to make sure changes in the current replica count will behave correctly with our changes.

**Comment by omersch381** [comment]
https://github.com/openstack-k8s-operators/openstack-operator/pull/1204 has been merged

**Comment by omersch381** [comment]
recheck podified-multinode-edpm-deployment-crc

**Pull request merged**